File: 2/container/Container.cpp
#include <algorithm>
#include "Container.h"

void Container::add(Matrix *matrix) {
    data[size++] = matrix;
}

Matrix *Container::get(int ind) {
    return data[ind];
}

void Container::set(int ind, Matrix *matrix) {
    data[ind] = matrix;
}

int Container::getSize() {
    return size;
}

void Container::sort() {
    for (int i = 0; i < size; i++) {
        for (int j = i + 1; j < size; j++) {
            if (getAverage(data[i]) > getAverage(data[j])) {
                std::swap(data[i], data[j]);
            }
        }
    }
}

double Container::getAverage(Matrix *matrix) {
    double sum = 0;
    for (int i = 0; i < matrix->size; i++) {
        for (int j = 0; j < matrix->size; j++) {
            sum += matrix->get(i, j);
        }
    }
    double sizeDouble = (double) matrix->size;
    return sum / (sizeDouble * sizeDouble);
}

File: 3/container/Container.py
import matrix.Matrix


class Container:

    def __init__(self):
        self.data = []

    def add(self, new_matrix):
        self.data.append(new_matrix)

    def get(self, index):
        return self.data[index]

    def get_size(self):
        return len(self.data)

    def set(self, index, new_matrix):
        self.data[index] = new_matrix

    def sort(self):
        for i in range(self.get_size()):
            for j in range(i + 1, self.get_size()):
                if self._get_average(self.data[i]) > self._get_average(self.data[j]) :
                    self.data[i], self.data[j] = self.data[j], self.data[i]

    def _get_average(self, cur_matrix):
        matrix_sum = 0
        for i in range(cur_matrix.size):
            for j in range(cur_matrix.size):
                matrix_sum += cur_matrix.get(i, j)
        return matrix_sum / (cur_matrix.size * cur_matrix.size)
